### PR TITLE
Host scaladoc on ghpages

### PR DESCRIPTION
**Pull request by Lewuathe**
## How to push the site

```
$ ./sbt unidoc ghpagesPushSite
```

see [here](https://wvlet.github.io/surface/latest/api/wvlet/surface/index.html)

### CODE REVIEW

**Comment by xerial** [comment]
@Lewuathe Looks good. Can we add 'unidoc ghpagesPushSite` command to sbt release steps so as not to forget releasing scaladoc?

File: build.sbt [hunk 1]
```diff
@@ -71,7 +72,19 @@ lazy val surfaceRoot =
     publishArtifact := false,
     publish := {},
     publishLocal := {}
-  ) aggregate(surfaceJVM, surfaceJS)
+  )
+  .enablePlugins(ScalaUnidocPlugin, GhpagesPlugin)
+  .settings(
+    name := "exclude JS",
```
Comment (xerial) [inline build.sbt]: Is this necessary? It will be overwritten in the following "ghpages settings" value.

Comment (Lewuathe) [inline build.sbt]: Yes, it's necessary. Without this compile failed due to duplicated class definition.
```
[error] /Users/sasakikai/dev/surface/surface/shared/src/main/scala/wvlet/surface/MethodSurface.scala:17: MethodSurface is already defined as trait MethodSurface
[error] trait MethodSurface {
[error]       ^
[error] /Users/sasakikai/dev/surface/surface/shared/src/main/scala/wvlet/surface/MethodSurface.scala:32: ClassMethodSurface is already defined as case class ClassMethodSurface
[error] case class ClassMethodSurface(mod: Int, owner: Surface, name: String, returnType: Surface, args: Seq[Parameter]) extends MethodSurface
```

I found the workaround [here]([error] /Users/sasakikai/dev/surface/surface/shared/src/main/scala/wvlet/surface/MethodSurface.scala:17: MethodSurface is already defined as trait MethodSurface
[error] trait MethodSurface {
[error]       ^
[error] /Users/sasakikai/dev/surface/surface/shared/src/main/scala/wvlet/surface/MethodSurface.scala:32: ClassMethodSurface is already defined as case class ClassMethodSurface
[error] case class ClassMethodSurface(mod: Int, owner: Surface, name: String, returnType: Surface, args: Seq[Parameter]) extends MethodSurface).

**Comment by xerial** [comment]
@Lewuathe Second thought. 
The latest Scaladoc is already available at: 
https://oss.sonatype.org/service/local/repositories/releases/archive/org/wvlet/surface_2.12/0.4/surface_2.12-0.4-javadoc.jar/!/wvlet/surface/index.html

And the link to it can be found from a badge in README.md. 
Simply updating a link to the scaladoc hosted on Sonatype might be sufficient rather than managing the latest api in gh-pages branch.

What do you think?

**Comment by Lewuathe** [comment]
> Simply updating a link to the scaladoc hosted on Sonatype might be sufficient rather than managing the latest api in gh-pages branch.

Sounds good to me. In case we want to host a surface homepage, releasing ghpage is good option.
But for now sonatype scaladoc is sufficient. I'll close this.

**Pull request closed without merging**